src/xshell_base.cpp

```diff
@@ -39,72 +39,30 @@ namespace xeus
         m_controller.bind(get_controller_end_point("shell"));
     }
 
-    xshell::~xshell()
+    xshell_base::~xshell_base()
     {
     }
 
-    std::string xshell::get_shell_port() const
+    std::string xshell_base::get_shell_port() const
     {
         return get_socket_port(m_shell);
     }
 
-    std::string xshell::get_stdin_port() const
+    std::string xshell_base::get_stdin_port() const
     {
         return get_socket_port(m_stdin);
     }
 
-    void xshell::run()
+    void xshell_base::run()
```

Comment:
It looks that this methods should forward the call to a pure virtual implementation method.